Fix non DSHOT protocols

klutvott123 opened the pull request:
Non DSHOT protocols are currently broken. At least multishot and oneshot.
Tested and confirmed working with all protocols except brushed and pwm, but that probably works too.

Fixes #11413, fixes #11363

ctzsnooze commented:
Thanks for finding the redundant line and solving the problem.  Amazing this wasn't reported earlier!

Tested with Bluejay DShot. found that dynamic idle works properly after these changes.  Switching profiles still switches the dynamic idle values correctly.  

Tested with BHeli-S to enable multishot, and all motors spin and handle idle settings correctly.

hydra commented:
> Thanks for finding the redundant line and solving the problem. Amazing this wasn't reported earlier!

Can you give details on the exact symptoms of failure?

Does that mean the only people testing BF 4.3 thus far have only been running DSHOT?  No users on pwm / oneshot / multishot or brushed motors anymore?

klutvott123 commented:
@hydra If dynamic idle is enabled, the code sets ```motorOutputLow``` to ```DSHOT_MIN_THROTTLE``` to allow zero motor drive. If dynamic idle is disabled the code would  add the dshot idle to that. The problem is that it did this for all protocols, not only dshot, and the endpoints aren't the same for all protocols. Setting the dshot idle like that is not necessary since it's all taken care of in ```initEscEndpoints()``` and we only have to override it for dshot if dynamic idle is enabled. 

The symptoms are that the motors wont spin or spins at the wrong rpm. Apparently very few are using protocols other than dshot and the issues linked to in the initial post are the only reports we have had about this so far.

hydra commented:
@klutvott123 thanks for the detailed description.  I felt that description of the issue was missing from the original description of the PR.  I guess when we say 'it's broken' we have to say how it's broken in order that reviewers can review a bit easier.

daleckystepan commented:
`motorOutputLow` seems to be directly connected to DSHOT itself. PR propose to tied it with DSHO transitively through `dynIdleMinRps`. I would propose to do it in the same manner as `dynIdleMinRps`. Maybe we can incorporate also `mixerRuntime.feature3dEnabled` and set it explicitly `motorOutputLow` to `0` if disabled (Hopefully I understand it correctly. It is a little bit hidden now so space to make it clearer).

haslinghuis commented:
@daleckystepan agree we should simplify, but this is a simple bugfix.

klutvott123 commented:
Added a comment to explain what's going on. Also added a call to ```initEscEndpoints()``` for ```MSP_SET_PID_ADVANCED``` to allow enabling and disabling dynamic idle without rebooting. This is in line with how it's already done for changind pid profile for example. This should now work exactly as originally intended.

@daleckystepan Yes, we could probably do some refactoring here, but right now I just want to make it work as it should.
